cmd/iceberg/main.go

```diff
@@ -0,0 +1,337 @@
+
+func main() {
+	args, err := docopt.ParseArgs(usage, os.Args[1:], iceberg.Version())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	cfg := struct {
+		List     bool `docopt:"list"`
+		Describe bool `docopt:"describe"`
+		Schema   bool `docopt:"schema"`
+		Spec     bool `docopt:"spec"`
+		Uuid     bool `docopt:"uuid"`
+		Location bool `docopt:"location"`
+		Props    bool `docopt:"properties"`
+		Drop     bool `docopt:"drop"`
+		Files    bool `docopt:"files"`
+		Rename   bool `docopt:"rename"`
+
+		Get    bool `docopt:"get"`
+		Set    bool `docopt:"set"`
+		Remove bool `docopt:"remove"`
+
+		Namespace bool `docopt:"namespace"`
+		Table     bool `docopt:"table"`
+
+		RenameFrom string `docopt:"<from>"`
+		RenameTo   string `docopt:"<to>"`
+
+		Parent   string `docopt:"PARENT"`
+		Ident    string `docopt:"IDENTIFIER"`
+		TableID  string `docopt:"TABLE_ID"`
+		PropName string `docopt:"PROPNAME"`
+		Value    string `docopt:"VALUE"`
+
+		Catalog string `docopt:"--catalog"`
+		URI     string `docopt:"--uri"`
+		Output  string `docopt:"--output"`
+		History bool   `docopt:"--history"`
+		Cred    string `docopt:"--credential"`
+	}{}
+
+	if err := args.Bind(&cfg); err != nil {
+		log.Fatal(err)
+	}
+
+	var output Output
+	switch strings.ToLower(cfg.Output) {
+	case "text":
+		output = text{}
+	case "json":
+		fallthrough
+	default:
+		log.Fatal("unimplemented output type")
+	}
+
+	opts := []catalog.Option{}
+	if len(cfg.Cred) > 0 {
+		opts = append(opts, catalog.WithCredential(cfg.Cred))
+	}
+
+	var cat catalog.Catalog
+	switch catalog.CatalogType(cfg.Catalog) {
+	case catalog.REST:
+		if cat, err = catalog.NewRestCatalog("rest", cfg.URI, opts...); err != nil {
+			log.Fatal(err)
+		}
+	default:
+		log.Fatal("unrecognized catalog type")
+	}
+
+	switch {
+	case cfg.List:
+		list(output, cat, cfg.Parent)
+	case cfg.Describe:
+		entityType := "any"
+		if cfg.Namespace {
+			entityType = "ns"
+		} else if cfg.Table {
+			entityType = "tbl"
+		}
+
+		describe(output, cat, cfg.Ident, entityType)
+	case cfg.Schema:
+		tbl := loadTable(output, cat, cfg.TableID)
+		output.Schema(tbl.Schema())
+	case cfg.Spec:
+		tbl := loadTable(output, cat, cfg.TableID)
+		output.Spec(tbl.Spec())
+	case cfg.Location:
+		tbl := loadTable(output, cat, cfg.TableID)
+		output.Text(tbl.Location())
+	case cfg.Uuid:
+		tbl := loadTable(output, cat, cfg.TableID)
+		output.Uuid(tbl.Metadata().TableUUID())
+	case cfg.Props:
+		properties(output, cat, propCmd{
+			get: cfg.Get, set: cfg.Set, remove: cfg.Remove,
+			namespace: cfg.Namespace, table: cfg.Table,
+			identifier: cfg.Ident,
+			propname:   cfg.PropName,
+			value:      cfg.Value,
+		})
+	case cfg.Rename:
+		_, err := cat.RenameTable(context.Background(),
+			catalog.ToRestIdentifier(cfg.RenameFrom), catalog.ToRestIdentifier(cfg.RenameTo))
+		if err != nil {
+			output.Error(err)
+			os.Exit(1)
+		}
+
+		output.Text("Renamed table from " + cfg.RenameFrom + " to " + cfg.RenameTo)
+	case cfg.Drop:
+		switch {
+		case cfg.Namespace:
```

Comment:
given that we can drop a namespace, should there be an option to create a namespace?